runtime/string.inl

```diff
@@ -820,10 +819,113 @@ bool string::is_int() const {
   return php_is_int(p, size());
 }
 
+bool string::is_numeric_as_php8() const {
+  const char *s = c_str();
```

Comment:
Let's make separate function like `is_numeric_impl` and get rid of copypaste?

Reply:
It seems an excessive refactoring, since php7-implementation will be dropped in the near future

Reply:
ok, seems sensible